src/zenml/zen_stores/schemas/pipeline_run_schemas.py

```diff
@@ -613,6 +647,8 @@ def update(self, run_update: "PipelineRunUpdate") -> "PipelineRunSchema":
                 if run_update.status_reason:
                     self.status_reason = run_update.status_reason
 
+            self.in_progress = self._check_if_run_in_progress()
```

Comment:
Somehow I'm still thinking we need to optimize this and not call it as often, because the _check_if_run_in_progress is essentially the most DB heavy operation that we're doing in all of the server. For example, do we need to call this every time, or is it enough to call it if the pipeline run status changes?